.github/workflows/packer-docker.yaml

```diff
@@ -0,0 +1,117 @@
+name: packer
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        type: string
+        required: false
+        default: '["ubuntu-latest"]'
+      ENVIRONMENTS:
+        description: '{"environments":[{"name":"dev"}, {"name":"qa"}]}'
+        required: true
+        type: string
+      IMAGE:
+        description: "Image to build Packer with"
+        required: true
+        type: string
+        default: "ghcr.io/xenitab/github-actions/tools:2020.12.6"
+      RESOURCE_GROUP_NAME_SUFFIX:
+        description: "Resource group name suffix to create VM image in"
+        required: true
+        type: string
+      PACKER_TEMPLATE_REPO:
+        description: "Repo to the packer template from"
+        required: true
+        type: string
+      PACKER_TEMPLATE_REPO_BRANCH:
+        description: "Branch to the packer template from"
+        required: true
+        type: string
+      PACKER_TEMPLATE_FILE:
+        description: "File path for the packer template"
+        required: true
+        type: string
+    secrets:
+      AZURE_CREDENTIALS_DEV:
+        required: false
+      AZURE_CREDENTIALS_QA:
+        required: false
+      AZURE_CREDENTIALS_PROD:
+        required: false
+
+jobs:
+  set_env_matrix:
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    runs-on: ${{fromJSON(inputs.runs-on)}}
+    steps:
+      - name: Set matrix for environment
+        id: set-matrix
+        run: |
+          set -e
+          echo '${{ inputs.ENVIRONMENTS }}' | jq .
+          MATRIX=$(echo '${{ inputs.ENVIRONMENTS }}' | jq -c .)
+          echo "::set-output name=matrix::${MATRIX}"
+
+  build:
+    needs: set_env_matrix
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.set_env_matrix.outputs.matrix)}}
+    runs-on: ${{fromJSON(inputs.runs-on)}}
+    steps:
+      - name: Create Azure credentials secret name
+        id: creds_env
+        run: |
+          AZURE_CREDS_TMP=AZURE_CREDENTIALS_${{ matrix.environments.name }}
+          AZURE_CREDS_ENV=$(echo $AZURE_CREDS_TMP |tr '[:lower:]' '[:upper:]')
+          echo "::set-output name=azure_creds_env::$AZURE_CREDS_ENV"
+
+      - name: Azure Login
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets[steps.creds_env.outputs.azure_creds_env] }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Packer build
+        env:
+          ENV: ${{ matrix.environments.name }}
+          TMP_DIR: ${{ runner.temp }}
+          IMAGE: ${{ inputs.IMAGE }}
+          RESOURCE_GROUP: "rg-${{ matrix.environments.name }}-we-${{ inputs.RESOURCE_GROUP_NAME_SUFFIX }}"
```

Comment:
I don't think that we always can assume that we are in region `we` I would add this as a variable.